src/reflect/value.go

```diff
@@ -1302,25 +1335,47 @@ func convertOp(src Value, typ Type) (Value, bool) {
 		*/
 
 	case Slice:
-		if typ.Kind() == String && !src.typecode.elem().isNamed() {
-			rtype := typ.(*rawType)
-
-			switch src.Type().Elem().Kind() {
-			case Uint8:
-				return cvtBytesString(src, rtype), true
-			case Int32:
-				return cvtRunesString(src, rtype), true
+		switch rtype := typ.(*rawType); rtype.Kind() {
+		case Array:
+			if src.typecode.elem() == rtype.elem() && rtype.Len() <= src.Len() {
+				return Value{
+					typecode: rtype,
+					value:    (*sliceHeader)(src.value).data,
```

Comment:
Thank you. I'm fumbling in the dark with the flags, because it's hard to come up with tests to see if they're set correctly. The behavior of when a value is addressable, etc is quite confusing.

Reply:
Updated. Here's my test program:

```go
//go:build ignore

package main

import (
	"fmt"
	"reflect"
)

func main() {
	s := make([]byte, 8)
	for i := range s {
		s[i] = byte(i)
	}

	a := reflect.ValueOf(s).Convert(reflect.TypeFor[[8]byte]()).Interface().([8]byte)
	b := reflect.ValueOf(s).Convert(reflect.TypeFor[*[8]byte]()).Interface().(*[8]byte)

	fmt.Printf("&a: %p\n&b: %p\n&s: %p\n", a[:], b[:], s)
	fmt.Printf("a: %v\nb: %v\ns: %v\n", a[:], (*b)[:], s)
}
```

Results:

```console
$ go run ./test.go
&a: 0xc00009a048
&b: 0xc00009a040
&s: 0xc00009a040
a: [0 1 2 3 4 5 6 7]
b: [0 1 2 3 4 5 6 7]
s: [0 1 2 3 4 5 6 7]

$ tinygo run ./test.go
&a: 0x7f07aca106d0
&b: 0x7f07aca106b0
&s: 0x7f07aca106b0
a: [0 1 2 3 4 5 6 7]
b: [0 1 2 3 4 5 6 7]
s: [0 1 2 3 4 5 6 7]
```